crates/matrix-sdk/src/account.rs

```diff
@@ -1165,6 +1173,79 @@ impl Account {
         self.set_account_data(unstable_media_preview_config).await?;
         Ok(())
     }
+
+    /// Adds a recently used emoji to the list and uploads the updated
+    /// `io.element.recent_emoji` content to the global account data.
+    ///
+    /// Before updating the data, it'll fetch it from the homeserver, to make
+    /// sure the updated values are always used.
+    #[cfg(feature = "experimental-element-recent-emojis")]
+    pub async fn add_recent_emoji(&self, emoji: &str) -> Result<()> {
+        let Some(user_id) = self.client.user_id() else {
+            return Err(Error::AuthenticationRequired);
+        };
+        let mut recent_emojis = self.get_recent_emojis(true).await?;
+
+        let index = recent_emojis.iter().position(|(unicode, _)| unicode == emoji);
+
+        let count = if let Some(index) = index { recent_emojis.remove(index).1 } else { uint!(0) };
+
+        recent_emojis.insert(0, (emoji.to_owned(), count + uint!(1)));
```

Comment:
This doesn't seem to limit the number of emojis that are stored. The JS SDK implementation starts truncating at a count of 100. Should something similar happen here, too?

Reply:
Just to confirm: do you truncate only when retrieving items, or also when add items so the account data event doesn't grow indefinitely?